packages/flutter_tools/test/integration.shard/test_test.dart

```diff
@@ -198,6 +198,31 @@ void main() {
   testWithoutContext('flutter gold skips tests where the expectations are missing', () async {
     return _testFile('flutter_gold', automatedTestsDirectory, flutterTestDirectory, exitCode: isZero);
   });
+
+  testWithoutContext('flutter test should respect --serve-observatory', () async {
```

Comment:
Would any of these tests fail without your change, or would they all pass because the Dart SDK defaults to serving the observatory?